Code/src/main/java/nl/utwente/group10/ui/components/blocks/input/RGBBlock.java

```diff
@@ -0,0 +1,116 @@
+package nl.utwente.group10.ui.components.blocks.input;
+
+import com.google.common.collect.ImmutableList;
+
+import javafx.fxml.FXML;
+import javafx.scene.layout.Background;
+import javafx.scene.layout.BackgroundFill;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.CycleMethod;
+import javafx.scene.paint.LinearGradient;
+import javafx.scene.paint.Stop;
+import nl.utwente.group10.ghcj.GhciException;
+import nl.utwente.group10.ghcj.GhciSession;
+import nl.utwente.group10.ui.CustomUIPane;
+import nl.utwente.group10.ui.components.anchors.InputAnchor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * Block with three inputs that represent RGB values.
+ * The RGBBlock colors according to these inputs.
+ */
+public class RGBBlock extends DisplayBlock {
+    /** The InputAnchor to which the Red value should be connected. **/
+    private InputAnchor r;
+    
+    /** The InputAnchor to which the Green value should be connected. **/
+    private InputAnchor g;
+    
+    /** The InputAnchor to which the Blue value should be connected. **/
+    private InputAnchor b;
+
+    /** The Pane to paint with the currently inputed color. **/
+    @FXML private Pane well;
+    
+    /** Default Background to use when no values are inputed. **/
+    private Background defaultBackground;
+
+    public RGBBlock(CustomUIPane pane) {
+        super(pane, "RGBBlock");
+
+        r = super.getAllInputs().get(0);
+        g = new InputAnchor(this);
+        b = new InputAnchor(this);
+        inputSpace.getChildren().setAll(ImmutableList.of(r, g, b));
+        
+        // Spread them out over the top of the Block.
```

Comment:
I was unable to spread them out using a GridPane, concluded it was not worth the effort to spend too much time on it